pkg/registry/client.go

```diff
@@ -616,14 +616,22 @@ func (c *Client) Tags(ref string) ([]string, error) {
 		}
 
 		break
-
 	}
+	return extractVersionsFromRegistryTags(registryTags), nil
+}
 
+// extractVersionsFromRegistryTags parses a list of OCI tag strings returns a
+// list containing only those that are valid Helm chart version strings.
+// The returned list is ordered by semantic version with the newest version
+// first.
+// Semantic versions are *not* de-duplicated.
+// Versions may have a v-prefix and if there are two semantically equal versions
+// with and without a v-prefix, both will be returned in their same order that
+// they were supplied.
```

Comment:
After having discovered all these subtleties while unit-testing, but it's made me question whether this is the right approach.

Perhaps it'd be better to change `helm push` to always use the strict (non-v-prefixed) semver as the OCI tag,
but that would break backwards compatibility.